includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -128,11 +128,12 @@ public function get_wcpay_product_id( WC_Product $product, $test_mode = null ) :
 	 * return string       The item's WCPay product id.
 	 */
 	public function get_wcpay_product_id_for_item( string $type ) : string {
-		if ( ! get_option( self::get_wcpay_product_id_option() . '_' . $type ) ) {
-			$this->create_product_for_item_type( $type );
+		$sanitized_type = sanitize_key( str_replace( ' ', '_', trim( strtolower( $type ) ) ) );
+		if ( ! get_option( self::get_wcpay_product_id_option() . '_' . $sanitized_type ) ) {
+			$this->create_product_for_item_type( $sanitized_type );
 		}
 
-		return get_option( self::get_wcpay_product_id_option() . '_' . $type );
+		return get_option( self::get_wcpay_product_id_option() . '_' . $sanitized_type );
```

Comment:
Also `self::get_wcpay_product_id_option() . '_' . $sanitized_type` can be a local variable - another micro optimization.

Reply:
Thanks for sharing this opportunity to make the code cleaner following DRY! I saved this as a local variable `$option_key_name`. I hope this looks fine?

Reply:
Yep, looks good!